### PR TITLE
feat(kuma-cp): add proxytemplate to matched policies for inspect poli…

### LINKED ISSUE #3718: Tools for visibility: add support for ProxyTemplate
### Description

Today there is no support in API for ProxyTemplate

### PR DESCRIPTION
**Pull request by tharun208**
### Summary

Based on #3462 

This PR introduces `proxyTemplate` policies to inspect policies endpoints.

### Full changelog

* Add `SelectProxyTemplate` method to get the matching policy
* Fix invalid import name in the sample `ProxyTemplate` conf

### Issues resolved

Fixes #3718

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [x] Unit tests
- [ ] E2E tests
- [x] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: pkg/test/kds/samples/resources.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ var (
 			Match: map[string]string{"serivce": "*"},
 		}},
 		Conf: &mesh_proto.ProxyTemplate_Conf{
-			Imports: []string{"default-kuma-profile"},
+			Imports: []string{"default-proxy"},
```
Comment (lobkovilya) [inline pkg/test/kds/samples/resources.go:244]: Why this change? Does it somehow affect the tests?

File: pkg/xds/template/proxy_template_resolver.go [hunk 2]
```diff
@@ -74,3 +74,14 @@ func (s sequentialResolver) GetTemplate(proxy *model.Proxy) *mesh_proto.ProxyTem
 func SequentialResolver(r ...ProxyTemplateResolver) ProxyTemplateResolver {
 	return sequentialResolver(r)
 }
+
+func SelectProxyTemplate(dataplane *core_mesh.DataplaneResource, proxyTemplates []*core_mesh.ProxyTemplateResource) *core_mesh.ProxyTemplateResource {
```
Comment (lobkovilya) [inline pkg/xds/template/proxy_template_resolver.go:78]: That's good idea to extract the logic from `SimpleProxyTemplateResolver#GetTemplate` into separate function and use it to build `MatchedPolicies` object. But could you please refactor `SimpleProxyTemplateResolver#GetTemplate` method to use this new `SelectProxyTemplate` function to avoid code duplication?

File: pkg/core/xds/matched_policies.go [hunk 3]
```diff
@@ -27,7 +27,8 @@ type MatchedPolicies struct {
 	TrafficRoutes      RouteMap
 
 	// Dataplane -> Policy
-	TrafficTrace *core_mesh.TrafficTraceResource
+	TrafficTrace  *core_mesh.TrafficTraceResource
+	ProxyTemplate *core_mesh.ProxyTemplateResource
```
Comment (lobkovilya) [inline pkg/core/xds/matched_policies.go:31]: Please drop a comment that actual Envoy configuration is generated without taking into account this `ProxyTemplate` field (and the same for `TrafficRoutes` field).

File: pkg/test/kds/samples/resources.go (hunk 1 above)
Comment (tharun208) [reply pkg/test/kds/samples/resources.go:244]: yes, it is throwing a validation error on the inspect dataplane test suite

File: pkg/xds/template/proxy_template_resolver.go (hunk 2 above)
Comment (tharun208) [reply pkg/xds/template/proxy_template_resolver.go:73]: addressed

File: pkg/core/xds/matched_policies.go [hunk 4]
```diff
@@ -27,7 +27,9 @@ type MatchedPolicies struct {
 	TrafficRoutes      RouteMap
 
 	// Dataplane -> Policy
-	TrafficTrace *core_mesh.TrafficTraceResource
+	// Actual Envoy Configuration is generated without taking these policies into account
```
Comment (lobkovilya) [inline pkg/core/xds/matched_policies.go:30]: This comment should be applied only to the `ProxyTemplate` field (not to `TrafficTrace`). Also, you can add a similar comment for `TrafficRoute` field

Comment (tharun208) [reply pkg/core/xds/matched_policies.go:30]: Do you mean the same comment to be duplicated for `TrafficTrace`?

Comment (lobkovilya) [reply pkg/core/xds/matched_policies.go]: No, TrafficTraces are fine, aren't they? I'm talking about TrafficRoutes, you added this field [here](https://github.com/kumahq/kuma/pull/3735) and this field is used only for Inspect API, it's not taken into account by proxy generator

Comment (tharun208) [reply pkg/core/xds/matched_policies.go]: understood

**Review by lobkovilya — APPROVED** [review]
Thank you for the contribution, @tharun208! I'll merge as soon as it will pass CI

**Comment by tharun208** [comment]
> Thank you for the contribution, @tharun208! I'll merge as soon as it will pass CI

thanks @lobkovilya 🚀
